### PR TITLE
Core 16716

### PR DESCRIPTION
**Pull request by BUGinator86**
## Purpose

_Fix for problem, where a user can install an application multiple times in parallel._

JIRA issue: [CORE-16716](https://jira.reactos.org/browse/CORE-16716)

## Proposed changes

_Use a flag to check if the download/installation of the selected application is already in progress_

## TODO

_Fix const usages and cleanup the code in some cases_

### CODE REVIEW

File: base/applications/rapps/include/available.h [hunk 1]
```diff
@@ -84,6 +87,7 @@ class CAvailableApplicationInfo
     VOID SetLastWriteTime(FILETIME* ftTime);
 
 private:
+    BOOL m_isInstalling;
```
Comment (HBelusca) [inline base/applications/rapps/include/available.h:90]:
```suggestion
    BOOL m_IsInstalling;
```

File: base/applications/rapps/loaddlg.cpp [hunk 2]
```diff
@@ -612,6 +611,11 @@ unsigned int WINAPI CDownloadManager::ThreadFunc(LPVOID param)
         }
         else
         {
+            if (InfoArray[iAppId].pApplicationInfo->IsInstalling())
+            {
+                continue;
+            }
+            InfoArray[iAppId].pApplicationInfo->StartInstalling();
```
Comment (HBelusca) [inline base/applications/rapps/loaddlg.cpp:618]: Is this whole thing thread-safe?

**Comment by HBelusca** [comment]
```
In file included from ../src/base/applications/rapps/include/dialogs.h:3,
                 from ../src/base/applications/rapps/include/rapps.h:10,
                 from ../src/base/applications/rapps/available.cpp:11:
../src/base/applications/rapps/include/available.h: In constructor 'CAvailableApplicationInfo::CAvailableApplicationInfo(const CStringW&, AvailableStrings&)':
../src/base/applications/rapps/include/available.h:93:10: error: 'CAvailableApplicationInfo::m_HasInstalledVersion' will be initialized after [-Werror=reorder]
     BOOL m_HasInstalledVersion;
          ^~~~~~~~~~~~~~~~~~~~~
../src/base/applications/rapps/include/available.h:90:10: error:   'BOOL CAvailableApplicationInfo::m_IsInstalling' [-Werror=reorder]
     BOOL m_IsInstalling;
          ^~~~~~~~~~~~~~
../src/base/applications/rapps/available.cpp:19:1: error:   when initialized here [-Werror=reorder]
 CAvailableApplicationInfo::CAvailableApplicationInfo(const ATL::CStringW& sFileNameParam, AvailableStrings& AvlbStrings)
 ^~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

**Comment by binarymaster** [comment]
Why closed?

**Comment by BUGinator86** [comment]
> Why closed?

I found to many other bugs that relate to this one.
I'll try to fix them in my own repo and open a new PR after that, so I don't unnecessarily waste your time. :)

**Pull request closed without merging**